plugins/modules/ds8000_assign_host_port.py

```diff
@@ -0,0 +1,191 @@
+                return self.changed, self.failed
+        wwpns = self._get_all_the_host_ports_under_specific_host(name)
+        if host_port not in wwpns:
+            try:
+                client.update_host_port_change_host(
+                    port_id=host_port, host_name=name)
+                self.changed = True
+            except pyds8k.exceptions.BadRequest as bad_wwpn:
+                self.failed = True
+                if ':' in host_port:
+                    self.module.fail_json(
+                        msg="This WWPN {host_port} not found, remove ':' from the WWPN. "
+                        "bad WWPN: 10:00:00:00:C9:A1:BA:B2 "
+                        "good WWPN: 10000000C9A1BAB2. "
+                        "ERR: {error}".format(
+                            host_port=host_port,
+                            error=to_native(bad_wwpn)))
+                else:
+                    self.module.fail_json(
+                        msg="This WWPN {host_port} not found. "
+                        "ERR: {error}".format(
+                            host_port=host_port, error=to_native(bad_wwpn)))
+            except Exception as generic_exc:
+                self.failed = True
+                self.module.fail_json(
+                    msg="can't assign this {host_port} WWPN to the host {host_name}. "
+                    "ERR: {error}".format(
+                        host_port=host_port,
+                        host_name=name,
+                        error=to_native(generic_exc)))
+
+        return self.changed, self.failed
+
+    def _check_if_the_host_port_is_free(self, host_port):
+        hosts = self._get_all_hosts()
+        name = self.params['name']
+        for host in hosts:
+            if host != name:
+                current_host_wwpns = self._get_all_the_host_ports_under_specific_host(
+                    host)
+                if host_port in current_host_wwpns:
+                    return True, host
+        return False, None
+
+    def _get_all_the_host_ports_under_specific_host(self, hostname):
+        token = self.token
+        wwpns = []
+        headers = {
+            "Content-Type": "application/json",
+            "X-Auth-Token": "{token}".format(token=token)
+        }
+        host_ports_url = "/hosts/{host_name}/host_ports".format(
+            host_name=hostname)
+        response = costume_get_request(self.module, headers, host_ports_url)
+        res = json.loads(response.text)
+        for host_port in res['data']['host_ports']:
+            wwpns.append(host_port['wwpn'])
+        return wwpns
+
+
+def main():
+    argument_spec = ds8000_argument_spec()
+    argument_spec.update(
+        name=dict(type='str'),
+        host_port=dict(type='list', elements='str'),
+        force=dict(type='bool', default=False)
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+    )
+
+    pds8kh = PyDs8kHelper(module)
+
+    if not pds8kh._check_if_ds8000_host_exists():
```

Comment:
Use public functions only when accessing them from outside the class
Also, remove the `not` and swap the true/false clauses for removing a double negative situation.